scipy/sparse/_coo.py

```diff
@@ -556,31 +571,94 @@ def _add_dense(self, other):
         dtype = upcast_char(self.dtype.char, other.dtype.char)
         result = np.array(other, dtype=dtype, copy=True)
         fortran = int(result.flags.f_contiguous)
-        M, N = self._shape_as_2d
-        coo_todense(M, N, self.nnz, self.row, self.col, self.data,
-                    result.ravel('A'), fortran)
+        if self.ndim < 3:
```

Comment:
Same here, let's use the nd version for 1-dimension cases.

Reply:
Addressed at 69e8757.